sending/src/main/java/org/kin/sending/presenter/ContactDialogPresenterImpl.java

```diff
@@ -0,0 +1,115 @@
+package org.kin.sending.presenter;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.kin.sending.view.ContactDialogView;
+import org.kin.sendkin.core.base.BasePresenterImpl;
+import org.kin.sendkin.core.model.KinAccountUtils;
+import org.kin.sendkin.core.model.RecipientContact;
+
+import java.util.UUID;
+
+public class ContactDialogPresenterImpl extends BasePresenterImpl<ContactDialogView> implements ContactDialogPresenter {
+
+    private SendKinPresenter sendKinPresenter;
+    private @Nullable
+    UUID contactId;
+    private String name = "";
+    private String address = "";
+    private RecipientContact recipientContact;
+
+    public ContactDialogPresenterImpl(SendKinPresenter sendKinPresenter, @Nullable UUID contactId) {
+        this.sendKinPresenter = sendKinPresenter;
+        this.contactId = contactId;
+    }
+
+    @Override
+    public void onResume() {
```

Comment:
Please try to use onSaveInstanceState and its counterpart.

Reply:
will be done in the next branch